.github/workflows/check-documentation.yml

```diff
@@ -0,0 +1,38 @@
+name: Check Documentation
+
+on:
+  push:
+    branches: [main, future]
+  pull_request:
+    branches: [main, future]
+
+jobs:
+  check-documentation:
+    name: Check documentation
+    runs-on: ubuntu-latest
+    env:
+      GO111MODULE: on
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: "1.18"
+
+      - name: generate documentation
+        run: go generate
+
+      - name: Verify Changed files
+        uses: tj-actions/verify-changed-files@v10.1
+        id: verify-changed-files
+        with:
+          files: |
+             docs
+
+      - name: Run step only when files change.
+        if: steps.verify-changed-files.outputs.files_changed == 'true'
+        run: |
+          echo "Changed files: ${{ steps.verify-changed-files.outputs.changed_files }}" && exit 1
```

Comment:
Maybe we could make the message a bit more friendly. Maybe something like "Changes have been detected in the docs folder. This can happen if you haven't run tfplugindocs after making a change. Please run `go generate` and commit the documentation changes to resolve this. Changed files: ..."

Reply:
I'd also rename the step to something like "Ensure documentation is up to date", so other people are clear about this step.